raft/src/main/java/org/apache/kafka/raft/RaftConfig.java

```diff
@@ -68,8 +70,21 @@
                 CommonClientConfigs.RETRY_BACKOFF_MS_DOC)
             .define(QUORUM_VOTERS_CONFIG,
                 ConfigDef.Type.LIST,
-                null,
-                new ConfigDef.NonEmptyString(),
+                ConfigDef.NO_DEFAULT_VALUE,
+                new ConfigDef.Validator() {
+                    // Copy from `NonEmptyListValidator` in Connect
+                    @Override
+                    public void ensureValid(String name, Object value) {
+                        if (value == null || ((List) value).isEmpty()) {
```

Comment:
One thought I had is that users might want to remove `quorum.voters` from the configuration after initial bootstrapping. I'm ok with the stricter validation for now, but something we should keep in mind. There's even a part of me that wants to allow pluggable bootstrapping so that we could do cool tricks like use a separate service. I'll file a jira and we can discuss.

Reply:
Sg!